README.md

```diff
@@ -1,2 +1,73 @@
+
+This is an imagined deployment of the Nexus with this adaptation.
+It implies that we are missing functionality in the Nexus API for mirroring (part of) the Nexus component tree in a browser Infusion instance.
+It also opens some questions about the practice of programming applications that intermix Infusion components and local-to-the-browser UI components.
+
+### Addressing spreadsheets
+Google spreadsheets are adressed by a combination of a spreadsheetId, sheetName, and a range or coordinate in A1 notation.
+The spreadsheetId is the alphanumeric identifier found in the URL for a spreadsheet, e.g. in `https://docs.google.com/spreadsheets/d/1B2769NqCW1yBlP_eT6kQ5iRBM1cbYyhB_0paDgNT6dk/` the spreadsheetId is `1B2769NqCW1yBlP_eT6kQ5iRBM1cbYyhB_0paDgNT6dk`.
+The sheetName is the identifier of a single 'tab' within the spreadsheet. It can be set by the user. The default value for the single tab in a new spreadsheet is `Sheet1`.
+A coordinate is written as a combination of letters referring to the column and a number referring to the row, e.g. `A1` for the top left cell.
+A range is written as two coordinates with a colon between them, e.g. `A1:B2` is the square of four cells at the top left of a spreadsheet.
+The first coordinate in a range must be in a lower column or row than the second coordinate.
+In API functions, the sheetName and coordinate/range are combined in one string, written `{sheetName}!{coordinate/range}`, e.g. `Sheet1!A1:B2`
+
+## Things you can make with nexus-gsheets
+
+### make a live visualization of a spreadsheet
+
+### collect streamed data in a spreadsheet
+
+### set up fine-grained data sharing between two spreadsheets
+
+## development plan
+ - [x] read data from a spreadsheet
+ - [x] make a config file defining aliases for sheets of interest, token location, credentials location
+ - [x] write data to a spreadsheet
+ - [ ] create a sheet in a Nexus component tree
+ - [ ] create a super-basic browser-and-Nexus setup
+ - [ ] support interactive token creation
+ - [ ] figure out how to package Nexus adapters as grades available to a Nexus instance on startup. Maybe copy Kettle's config functionality?
+ - [ ] set up unit testing, including a test mock of the google sheets API client object. 
+ - [ ] figure out how to distribute the demo. I assume it either needs to work with public sheets, or there should be a user dialog for setting up credentials and either providing a URL or naming the data to go in a sheet
+
+Do I depend on the nexus or not?
+If I want to start the Nexus server from here, yes.
+That means starting a kettle server that hosts a component tree which may be manipulated through the Nexus API.
+the existing demos don't actually depend on the nexus? They just assume one will be started elsewhere
+what is my ideal setup process?
+open the demo, possibly open a google spreadsheet
+
+It seems like we'd need to manually implement polling to get updates in response to sheet content changing.
```

Comment:
Looks like in the long term we probably could get value from the drive push notification API - https://developers.google.com/drive/api/v3/reference/changes/watch - the restriction that all it could do is call back to a publically exposed HTTP endpoint in the long run won't be too onerous given we will have to host some kind of publically usable service in any case.